### ISSUE TITLE
pipe in inline code is considered a column delimiter in standard flavor

### ISSUE THREAD

**Issue by varac** [rvben/rumdl#199]
Given this table:

```sh
❯ cat table.md 
# Markdown table example

| Name | value |
| ---- | ----- |
| foo |  `a|b` |
```

## standard flavor 

With the `stadard` flavor, the inline code is truncated and the pipe is wrongly considered as a column delimiter:

```sh
❯ cat .rumdl.toml 
[global]
flavor = "standard"

❯ rumdl check --diff table.md
--- table.md
+++ table.md (fixed)
@@ -2,4 +2,4 @@
 
 | Name | value |
 | ---- | ----- |
-| foo |  `a|b` |
+| foo | `a |

...
```

## mkdocs flavor 

With the `mkdocs` flavor, the tables remains unchanged: 

```sh 
❯ cat .rumdl.toml
[global]
flavor = "mkdocs"

❯ rumdl check --diff table.md

Success: No issues found in 1 file (49ms)
```

## Expected behaviour

The table should stay unchanged even when using the `standard` flavor.

**Comment by varac** [comment]
Somehow related: #165

**Comment by rvben** [comment]
In the CommonMark/GFM spec (which GitHub follows), pipes inside inline code **ARE** treated as cell delimiters in tables. This is 'by design' in GitHub's rendering engine for some reason. 

I chose to fix this in a different way. If this happens the pipe in the inline code gets escaped to prevent messing up the content.:

**Before (corrupted):**
```markdown
| foo | `a|b` |  →  | foo | `a |
```

**After (preserved):**
```markdown
| foo | `a|b` |  →  | foo | `a\|b` |
```

This ensures the rendering is correct on GitHub and it keeps the valid table structure.

**Comment by varac** [comment]
Thanks @rvben  !
